platforms/hermes-javascript/README.md

```diff
@@ -0,0 +1,654 @@
+    })
+
+    /*
+        Now this is all for the basics, but for managing a dialog session
+        using .on / .off / .once / .publish is actually not the best way!
+
+        The Dialog API also exposes a small wrapper that make these operations much easier,
+        and it is strongly recommended to use this wrapper instead!
+
+        See below for an example on how to build a dialog flow tree using this API.
+    */
+
+    /*
+        The goal is to register the following dialog paths:
+        A
+        ├── B
+        │   └─ D
+        └── C
+        In plain words, intent 'A' starts the flow, then restrain the next intents to 'B' or 'C'.
+        If 'B' is the next intent detected, then next intent must be 'D' (and end the flow after 'D').
+        If it was 'C', end the flow.
+    */
+    dialog.flow('A', (msg, flow) => {
+
+        console.log('Intent A received. Session started.')
+
+        /*
+            At each step of the dialog flow, you have the choice of
+            registering the next intents, or end the flow.
+
+            We then subscribe to both intent B or C so that the dialog
+            flow will continue with either one or the other next.
+        */
+
+        // Mark intent 'B' as one of the next dialog intents. (A -> B)
+        flow.continue('B', (msg, flow) => {
+            console.log('Intent B received. Session continued.')
+
+            // Mark intent 'D'. (A -> B -> D)
+            flow.continue('D', (msg, flow) => {
+                console.log('Intent D received. Session is ended.')
+                flow.end()
+                return 'Finished the session with intent D.'
+            })
+
+            // Make the TTS say that.
+            return 'Continue with D.'
+        })
+
+        // Mark intent 'C' as one of the next dialog intents. (A -> C)
+        flow.continue('C', (msg, flow) => {
+            const slotValue = msg.slots[0].value.value
+            console.log('Intent C received. Session is ended.')
+            flow.end()
+            return 'Finished the session with intent C having value ' + slotValue + ' .'
+        })
+
+        // The continue / end message options (basically text to speech)
+        // If the return value is a string, then it is equivalent to { text: '...' }
+        return 'Continue with B or C.'
+    })
+
+})
+```
+
+## API
+
+### Context loop
+
+An hermes client should implement a context loop that will prevent the program from exiting.
+
+#### Using withHermes
+
+```js
+const { withHermes } = require('hermes-javascript')
+
+// See the Hermes class documentation below for available options.
+const hermesOptions = { /* ... */ }
+
+/*
+The withHermes function automatically sets up the context loop.
+
+Arguments:
+   - hermes is a freshly created instance of the Hermes class
+   - call done() to exit the loop and destroy() the hermes instance
+*/
+withHermes((hermes, done) => {
+    /* ... */
+}, hermesOptions)
+```
+
+#### Using the keepAlive tools
+
```

Comment:
maybe explicit the doc on this a little more ? this is what's used to implement the `done` callback in `withHermes` if I understand correctly, what is the use case to use it directly for an user of hermes (if there is non maybe don't put it in the readme ? )

Reply:
Added a comment about when it should be used.